### ISSUE TITLE
Key problem

### ISSUE THREAD

**Issue by GitHub-TXZ** [MIC-DKFZ/nnUNet#1775]
Dear Fabian, I have two questions:
1) Will manually setting target spacing and patch size significantly affect the performance of the model?
2) When I continued training in the first and second epoch settings respectively, his curve is as shown in the figure below. The first continued training only added one epoch, and the second continued training setting added 10 epochs. This Is the learning rate decreasing like this because the total number of epochs is different?

**Comment by GitHub-TXZ** [comment]
![image](https://github.com/MIC-DKFZ/nnUNet/assets/73587759/7124a981-9920-4049-9d81-6a51ae7c9cd0)

**Comment by TaWald** [comment]
> Will manually setting target spacing and patch size significantly affect the performance of the model?

Yes. choosing the target spacing and patch size is problem dependent and can have a large influence.

> When I continued training in the first and second epoch settings respectively, his curve is as shown in the figure below. The first continued training only added one epoch, and the second continued training setting added 10 epochs. This Is the learning rate decreasing like this because the total number of epochs is different?

Whenever you train your learning rate is defined by polylr schedule. Now when you switch the amount of total training epochs the learning rate schedule will look different, hence your weird peak, as the learning rate schedule is not set up to be continuous. Check the LR schedluer definition [here](https://github.com/MIC-DKFZ/nnUNet/blob/2bc504de420fb64f130bfe22ce9867cbad2b9d4a/nnunetv2/training/lr_scheduler/polylr.py#L4)

**Comment by GitHub-TXZ** [comment]
so target spcing and patch size are so important hyperparameters, is nnunet the&nbsp;optimalist？


啊哈哈
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "MIC-DKFZ/nnUNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月8日(星期三) 晚上7:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [MIC-DKFZ/nnUNet] Key problem (Issue #1775)





  
Will manually setting target spacing and patch size significantly affect the performance of the model?
  
Yes. choosing the target spacing and patch size is problem dependent and can have a large influence.
  
When I continued training in the first and second epoch settings respectively, his curve is as shown in the figure below. The first continued training only added one epoch, and the second continued training setting added 10 epochs. This Is the learning rate decreasing like this because the total number of epochs is different?
  
Whenever you train your learning rate is defined by polylr schedule. Now when you switch the amount of total training epochs the learning rate schedule will look different, hence your weird peak, as the learning rate schedule is not set up to be continuous. Check the LR schedluer definition here

**Comment by TaWald** [comment]
The question what is optimal is not known and still subject to debate. Spacing wise nnU-Net chooses it given the dataset statistics, e.g. median spacing. For patch sizes the same applies, but it costs a lot of VRAM so there is a trade-off between multiple points: Architecture - Batch-Size - Patch-Size 
nnU-Net itself is not optimal for the datasets, it just provides a robust starting point for you to start exploring yourself.

**Comment by GitHub-TXZ** [comment]
thanks a lot，nnunet my super hero



---Original---
From: "Tassilo ***@***.***&gt;
Date: Thu, Nov 9, 2023 16:00 PM
To: ***@***.***&gt;;
Cc: "Tan ***@***.******@***.***&gt;;
Subject: Re: [MIC-DKFZ/nnUNet] Key problem (Issue #1775)




 
The question what is optimal is not known and still subject to debate. Spacing wise nnU-Net chooses it given the dataset statistics, e.g. median spacing. For patch sizes the same applies, but it costs a lot of VRAM so there is a trade-off between multiple points: Architecture - Batch-Size - Patch-Size
 nnU-Net itself is not optimal for the datasets, it just provides a robust starting point for you to start exploring yourself.
